### DISCUSSION TITLE
Plotting TPM counts for one crab over 3 days

### DISCUSSION THREAD

**Discussion by aspencoyle** [RobertsLab/resources#1206] [Q&A] [answered]
Sorry for the delay - here's all TPM counts for a single crab (specifically Crab A) over time. This particular crab was kept at ambient temperature throughout the course of the experiment

For background, this library was constructed by aligning the library to cbai_transcriptome_v4.0, which was constructed by BLASTing against the _C. opilio_ genome. In other words, this library should contain only _C. bairdi_ sequences.

[Crab_A_cbai_transcriptome_v4.0.txt](https://github.com/RobertsLab/resources/files/6456746/Crab_A_cbai_transcriptome_v4.0.txt)

**Comment by sr320** [comment]
Looking for anyone to plot these data and post here

**Comment by laurahspencer** [comment, marked as answer]
Genes have been filtered to retain those with 5+ reads across all 3 time points 

![image](https://user-images.githubusercontent.com/17264765/117860827-978cb280-b245-11eb-9ca3-6cdd97e8d458.png)

Code: 
[pheatamp_cbairdi.zip](https://github.com/RobertsLab/resources/files/6461019/pheatamp_cbairdi.zip)

**Reply by sr320** [reply]
I like it! to me there are clear clusters!

**Reply by sr320** [reply]
@laurahspencer is there a way to pull genelists from this based on clustering?

**Reply by kubu4** [reply]
@grace-ac You did something like this with the crab clustering, right? Can you point to some code to pull info from clusters?

**Reply by grace-ac** [reply]
Yep! here's a script: https://github.com/RobertsLab/paper-tanner-crab/blob/master/scripts/heatmaps-DEGs_in-singlecrab_over-time.Rmd

**Comment by laurahspencer** [comment]
Yes the script in my post should let you pull gene lists from clusters - I
didn’t dive too deep into it, but you define the number of clusters you
want, then extract gene lists from there

**Comment by laurahspencer** [comment]
Apparently the script I included in that zip folder was the wrong one!  Here's the code: 


### Load libraries and source scripts 

```{r, message=FALSE, warning=FALSE, results=FALSE}
list.of.packages <- c("pheatmap", "tidyverse") #add new libraries here 
new.packages <- list.of.packages[!(list.of.packages %in% installed.packages()[,"Package"])]
if(length(new.packages)) install.packages(new.packages)

# Load all libraries 
lapply(list.of.packages, FUN = function(X) {
  do.call("require", list(X)) 
})
```

## Load count data 

```{r}
(data <- read.table(file = "Crab_A_cbai_transcriptome_v4.0.txt"))

# Drop any genes with less than 5 counts across all 3 samples 
keep <- rowSums(data) >= 5
data.f <- data[keep,]
print(paste("# genes remaining after pre-filtering:", nrow(data.f)))
print(paste("# of genes dropped:", nrow(data) - nrow(data.f), sep=" "))
```
## Construct heat map 

```{r}
pheatmap(data.f, cluster_rows=TRUE, show_rownames=F, na.rm=TRUE, cluster_cols = FALSE, scale="row", main = "gene counts by day", fontsize = 9) 
```

```{r}
# extract gene clusters 
out <- pheatmap(data.f, cluster_rows=TRUE, show_rownames=F, na.rm=TRUE, cluster_cols = FALSE, scale="row", main = "gene counts by day", fontsize = 9) 
hc <-out$tree_row
lbl <- cutree(hc, 4) # split gene dendrogram in 4 groups
which(lbl==1) %>% names() # grab genes of first group
```

# Extract counts for the 4 gene clusters 
```{r}
data.f[which(lbl==1) %>% names(),]
data.f[which(lbl==2) %>% names(),]
data.f[which(lbl==3) %>% names(),]
data.f[which(lbl==4) %>% names(),]
```

**Reply by laurahspencer** [reply]
You can split the dendogram into more/less clusters above at the `cutree` function. 
Also, you can generate heat maps for gene sets in each cluster, like this: 

```
clust1 <- data.f[which(lbl==1) %>% names(),]
clust2 <- data.f[which(lbl==2) %>% names(),]
clust3 <- data.fwhich(lbl==3) %>% names(),]
clust4 <- data.f[which(lbl==4) %>% names(),]

pheatmap(clust1, cluster_rows=TRUE, show_rownames=F, na.rm=TRUE, 
cluster_cols = FALSE, scale="row", main = "gene counts by day, cluster 1", fontsize = 9) 
```
